crates/cli/cli-components/src/impls/commands/queries/connection_end.rs

```diff
@@ -0,0 +1,89 @@
+use core::marker::PhantomData;
+
+use cgp_core::prelude::*;
+use hermes_relayer_components::build::traits::builders::chain_builder::CanBuildChain;
+use hermes_relayer_components::chain::traits::queries::chain_status::CanQueryChainHeight;
+use hermes_relayer_components::chain::traits::queries::connection_end::CanQueryConnectionEnd;
+use hermes_relayer_components::chain::traits::types::chain_id::HasChainId;
+use hermes_relayer_components::chain::traits::types::connection::HasConnectionEndType;
+use hermes_relayer_components::chain::traits::types::height::HasHeightType;
+use hermes_relayer_components::multi::traits::chain_at::HasChainTypeAt;
+use hermes_relayer_components::multi::types::index::Index;
+
+use crate::traits::build::CanLoadBuilder;
+use crate::traits::command::CommandRunner;
+use crate::traits::output::CanProduceOutput;
+use crate::traits::parse::CanParseArg;
+
+pub struct RunQueryConnectionEndCommand;
+
+#[derive(Debug, clap::Parser, HasField)]
+pub struct QueryConnectionEndArgs {
+    #[clap(
+        long = "chain",
+        required = true,
+        value_name = "CHAIN_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chain to query"
+    )]
+    chain_id: String,
+
+    #[clap(
+        long = "connection",
+        visible_alias = "conn",
+        required = true,
+        value_name = "CONNECTION_ID",
+        help_heading = "REQUIRED",
+        help = "Identifier of the connection to query"
+    )]
+    connection_id: String,
+
+    #[clap(
+        long = "height",
+        value_name = "HEIGHT",
+        help = "Height of the state to query. Leave unspecified for latest height."
+    )]
+    height: Option<String>,
+}
+
+impl<App, Args, Build, Chain, Counterparty> CommandRunner<App, Args>
+    for RunQueryConnectionEndCommand
+where
+    App: CanLoadBuilder<Builder = Build>
+        + CanProduceOutput<Counterparty::ConnectionEnd>
+        + CanParseArg<Args, symbol!("chain_id"), Parsed = Chain::ChainId>
+        + CanParseArg<Args, symbol!("connection_id"), Parsed = Chain::ConnectionId>
+        + CanParseArg<Args, symbol!("height"), Parsed = Option<Chain::Height>>
```

Comment:
We need to add the following wiring to [`HermesParserComponents`](https://github.com/informalsystems/hermes-sdk/blob/668e55080bef8e13ffffbeceba49d5be74de0ff8/crates/cli/cli/src/contexts/app.rs#L145) to enable the parsing of the given fields.

```
        (QueryConnectionEndArgs, symbol!("chain_id")): ParseFromString<ChainId>,
        (QueryConnectionEndArgs, symbol!("connection_id")): ParseFromString<ConnectionId>,
        (QueryConnectionEndArgs, symbol!("height")): ParseFromOptionalString<Height>,
```